test/Optimizer/Miscellaneous_MaxInterpret.js

```diff
@@ -1071,3 +1071,23 @@ function test66() {
 }
 WScript.Echo("test66: " + test66());
 WScript.Echo("test66: " + test66());
+
+(function test67Runner(){
+    function test67(a,b,x){
+        var i = a[0];
+        var j = b[0];
+        var t = a;
+        for (var p = 0; p < x; p++) {
+            if (p > 2) {
+                t = b;
+                b = a;
+            }

```

Comment:
If we don't add compenstation in the prepass, won't we end up with a null lengthSym in this case in the merge block ?

Reply:
only in the prepass. it will be fixed up in the main pass

Reply:
After some more offline discussions with @meg-gupta, we concluded that, yes, `b` will end with a null lengthSym in the block after the if block. This, when merged with the lengthSym from the landing pad, will result in a null length sym for `b` in the main pass.

While this is not ideal, its also not a regression from the pre-AggressiveValueTransfer days. I'll keep the fix as is for now, but keep a note of the possible optimization opportunity.